src/System.Memory/src/System/SpanHelpers.T.cs

```diff
@@ -409,6 +576,24 @@ public static int LastIndexOfAny<T>(ref T searchSpace, T value0, T value1, T val
             return length + 7;
         }
 
+        public static int LastIndexOfAny<T>(ref T searchSpace, int searchSpaceLength, ref T value, int valueLength)
+            where T : IEquatable<T>
+        {
+            Debug.Assert(searchSpaceLength >= 0);
+            Debug.Assert(valueLength >= 0);
+
+            if (valueLength == 0)
+                return 0;  // A zero-length sequence is always treated as "found" at the start of the search space.
+
+            int index = -1;
+            for (int i = 0; i < valueLength; i++)
+            {
+                var tempIndex = LastIndexOf(ref searchSpace, Unsafe.Add(ref value, i), searchSpaceLength);
+                if (tempIndex > index) index = tempIndex;
```

Comment:
Should this be `if ((uint)tempIndex < (uint)index)` just like IndexOfAny? If so, can you please see if we can add a test case where not casting to uint fails to give the right answer?

Reply:
I think, no.
Let's say we try to find last index of any of [1, 3] values in [3, 1] span.
- After search for value 1, index will be -1 and tempIndex is 1. In both cases index will be set to 1.
- After search for value 3, index will be 1, and tempIndex is 0. In case of `tempIndex > index` check index will stay 1, but in case of `(uint)tempIndex < (uint)index` check index will be set to 0 - which is **not correct**.

Reply:
Ah yes, my bad. I reverted the inequality be accident. I meant, should we cast both to uint like IndexOfAny?

i.e. Should LastIndexOfAny be:
`if ((uint)tempIndex >(uint)index)` instead of `if (tempIndex > index)`

Reply:
No. In case of not found value tempIndex will be -1, and "converting" to uint will lead to biggest value (uint.MaxValue), so that `if ((uint)tempIndex >(uint)index)` check will pass and index will be set to -1.